### PR TITLE
Feature/38377 jest for admin tests

### PR DESCRIPTION
**Pull request by GuyHarwood**
- [x] refactor jasmine dependencies
- [x] implement default config and exclusions
- [x] ensure coverage excludes relevant paths
- [x] add support for integration tests

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -25,17 +25,23 @@ jobs:
...
           key: v1-node-tslib-deps-{{ checksum "tslib/yarn.lock" }}
       - run: cd admin; yarn install --frozen-lockfile
+      - run:
+          name: Install JUnit coverage reporter
+          command: cd admin; yarn add --dev jest-junit
+      - run:
+          name: Run tests with JUnit as reporter
+          command: cd admin; ./node_modules/.bin/jest --ci --runInBand --reporters=default --reporters=jest-junit --collectCoverage=true
+          environment:
+            JEST_JUNIT_OUTPUT_DIR: "reports/junit/js-test-results.xml"
```
Comment (GuyHarwood) [inline .circleci/config.yml]: reordering the sequence here to optimise for fast failing, so it only performs the install before the test run on each app.

File: admin/package.json [hunk 2]
```diff
@@ -4,19 +4,18 @@
...
     "seed-sql": "node ./data/sql/seed-sql.js",
     "migrate-and-seed-sql": "yarn migrate-sql && yarn seed-sql",
     "postinstall": "gulp build",
     "sasswatch": "gulp watch",
-    "sql-integration-test": "nyc -r html ./tests-integration/run-jasmine-integration.js",
     "start": "node ./server.js",
     "startd": "yarn migrate-sql && node ./server.js",
-    "test": "yarn lint && yarn run nyc -r html ./run-jasmine.js",
-    "client-test": "karma start karma.conf.js",
-    "testDebug": "node $NODE_DEBUG_OPTION ./node_modules/.bin/jasmine",
```
Comment (GuyHarwood) [inline admin/package.json]: i have aligned the script names here

File: admin/package.json [hunk 3]
```diff
@@ -109,38 +105,42 @@
...
     "madge": "^3.2.0",
     "node-mocks-http": "^1.6.6",
-    "nyc": "^14.1.1",
     "proxyquire": "^2.1.0",
     "sinon": "^7.1.1",
     "standard": "^14.3.1",
     "tslint": "^5.19.0",
     "typescript": "^3.5.3"
   },
-  "nyc": {
```
Comment (GuyHarwood) [inline admin/package.json]: coverage now done by jest

File: admin/spec/back-end/controllers/accessibility-statement.spec.js [hunk 4]
```diff
@@ -5,7 +5,7 @@
 const httpMocks = require('node-mocks-http')
 
 describe('accessibility-statement page controller', () => {
-  it('should render the initial accessibility-statement page', async (done) => {
```
Comment (GuyHarwood) [inline admin/spec/back-end/controllers/accessibility-statement.spec.js]: the rest of the PR is now just removing the done param and calls which were necessary to support async in jasmine.  there is a lot of it.

**Comment by GuyHarwood** [comment]
@jon-shipley @GeorgeChatzigiannis final reviews please

File: admin/package.json [hunk 5]
```diff
@@ -4,19 +4,18 @@
...
-    "sql-integration-test": "nyc -r html ./tests-integration/run-jasmine-integration.js",
     "start": "node ./server.js",
     "startd": "yarn migrate-sql && node ./server.js",
-    "test": "yarn lint && yarn run nyc -r html ./run-jasmine.js",
-    "client-test": "karma start karma.conf.js",
-    "testDebug": "node $NODE_DEBUG_OPTION ./node_modules/.bin/jasmine",
+    "test": "yarn lint && yarn jest",
+    "test:watch": "yarn jest --watch",
+    "test:integration": "yarn jest --config ./tests-integration/jest.integration.config.js",
+    "test:frontend": "karma start karma.conf.js",
```
Comment (GeorgeChatzigiannis) [inline admin/package.json]: should we change frontend to run on jest as well?

Comment (GuyHarwood) [reply admin/package.json]: Karma supports auto reload i believe, and any conversion to jest would need to be in a separate ticket

**Pull request merged**